src/utils/d2c_messaging/src/d2c_messaging.c

```diff
@@ -456,7 +491,7 @@ void ADUC_D2C_Messaging_Uninit()
     if (s_core_initialized)
     {
         // Cancel pending messages
-        for (int i = 0; i < ADUC_D2C_Message_Type_Max; i++)
+        for (int i = 0; i < ADUC_D2C_Message_Type_Max && s_messageProcessingContext[i].initialized; i++)
         {
             pthread_mutex_lock(&s_messageProcessingContext[i].mutex);
```

Comment:
Should this be locking s_pendingMessageStore[i].mutex and lock again on line 502 for s_messageProcessingContext[i].mutex -- the idea being separate mutex for each granular data array?

Reply:
Is s_pendingMessageStoreMutex being used as transaction lock? Instead, if possible, it should only be locking access to s_pendingMessageStore array. For s_core_initialized, lock can be avoided if can guarantee init and unint are called on main thread only, or else have yet another lock just for accessing s_core_initialized.